ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -297,20 +253,44 @@ noLoPBucket =
     , lbGrantToken = pure ()
     }
 
+data JumpingGovernor m blk = JumpingGovernor {
+    jgNextInstruction   :: !(m (JumpingGovernor.Instruction blk))
+  , jgProcessJumpResult :: !(JumpingGovernor.JumpResult blk -> m ())
+  }
+  deriving stock (Generic)
```

Comment:
I never quite know how to document those things, because they are only the hooks. So if you want to fully change their semantics, you're free to do so. In practice, there will just be two types of hooks, though, either _the_ ChainSync jumping ones or no-op versions for when CSJ is unwanted/unneeded. I gave it a try in 56b09b9e13af4534272a17ef73155a5509213b17; is that enough?

Reply:
I would like to have the story of how these hooks are supposed to be used by the ChainSync client. Otherwise, how do we know that they aren't omitted in needed places?